core/commands/config.go

```diff
@@ -69,93 +73,66 @@ Set the value of the 'Datastore.Path' key:
 		cmdkit.BoolOption(configBoolOptionName, "Set a boolean value."),
 		cmdkit.BoolOption(configJSONOptionName, "Parse stringified JSON."),
 	},
-	Run: func(req cmds.Request, res cmds.Response) {
-		args := req.Arguments()
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		args := req.Arguments
 		key := args[0]
 
 		var output *ConfigField
-		defer func() {
-			if output != nil {
-				res.SetOutput(output)
-			} else {
-				res.SetOutput(nil)
-			}
-		}()
 
 		// This is a temporary fix until we move the private key out of the config file
 		switch strings.ToLower(key) {
 		case "identity", "identity.privkey":
-			res.SetError(fmt.Errorf("cannot show or change private key through API"), cmdkit.ErrNormal)
-			return
+			return fmt.Errorf("cannot show or change private key through API")
 		default:
 		}
 
-		r, err := fsrepo.Open(req.InvocContext().ConfigRoot)
+		ctx := env.(*oldcmds.Context)
+		r, err := fsrepo.Open(ctx.ConfigRoot)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 		defer r.Close()
 		if len(args) == 2 {
 			value := args[1]
 
-			if parseJSON, _, _ := req.Option(configJSONOptionName).Bool(); parseJSON {
+			if parseJSON, _ := req.Options[configJSONOptionName].(bool); parseJSON {
 				var jsonVal interface{}
 				if err := json.Unmarshal([]byte(value), &jsonVal); err != nil {
 					err = fmt.Errorf("failed to unmarshal json. %s", err)
-					res.SetError(err, cmdkit.ErrNormal)
-					return
+					return err
 				}
 
 				output, err = setConfig(r, key, jsonVal)
-			} else if isbool, _, _ := req.Option(configBoolOptionName).Bool(); isbool {
+			} else if isbool, _ := req.Options[configBoolOptionName].(bool); isbool {
 				output, err = setConfig(r, key, value == "true")
 			} else {
 				output, err = setConfig(r, key, value)
 			}
 		} else {
 			output, err = getConfig(r, key)
 		}
+
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
-	},
-	Marshalers: cmds.MarshalerMap{
-		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			if len(res.Request().Arguments()) == 2 {
-				return nil, nil // dont output anything
-			}
-
-			if res.Error() != nil {
-				return nil, res.Error()
-			}
-
-			v, err := unwrapOutput(res.Output())
-			if err != nil {
-				return nil, err
-			}
 
-			vf, ok := v.(*ConfigField)
-			if !ok {
-				return nil, e.TypeErr(vf, v)
+		return res.Emit(output)
```

Comment:
Same, this needs to use `EmitOnce`.

Reply:
@Stebalien I found a strange question, when I use `cmds.EmitOnce(output)` here, it will fail to pass `t0063-external.sh` testcase.  After analyzing the `t0063-external.sh`. I found that `ipfs config key value` won't remove `repo.lock` file successfully. However, `ipfs config key value` won't print any error or panic.

Reply:
Hm. That's not good. @magik6k or @kevina, want to look into this?